File: 0007_reverse_integer/reverse_integer.py
def reverse(x: int) -> int:
    """
    Time: O(n)
    Space: O(1)
    """
    s = [1, -1][x < 0]
    res = s * int(str(abs(x))[::-1])
    return res if res.bit_length() < 32 else 0

File: 0007_reverse_integer/reverse_integer.js
/**
 * Time:    O(n)
 * Space:   O(1)
 * @param   {number} x
 * @return  {number}
 */
const reverse = x => {
  const INT_MAX = 2**31 - 1
  const INT_MIN = -(2**31)
  const sign = Math.sign(x)
  let res = 0
  x = Math.abs(x)
  while (x) {
      res = res * 10 + x % 10
      x = Math.floor(x / 10)
  }
  return res < INT_MAX && res > INT_MIN ? res * sign : 0
}
